### PR TITLE
[WIP] Carp taming PART 2

### PR DESCRIPTION
**Pull request by unid15**
- When two mobs of the same species (human, vox, monkey, .....) tame a carp, the carp starts viewing all mobs of that species as its friends. It'll still attack mobs of other species (unless it's tamed by them)
- Dealing damage to hostile animals that are friendly to you has a major chance of making them angry at you (and your species).
- Fix for animals growing up while being held

_Race war now_

### CODE REVIEW

**Comment by gbasood** [comment]
I don't see ANY WAY THIS COULD GO WRONG :+1: 
Vox slums filled with carp pets when

**Comment by Exxion** [comment]
I feel like it should take more than two tamings to make them friendly to a whole species, but otherwise :+1:

**Comment by ghost** [comment]
Important question time.

Can you pet the carp?

**Comment by Exxion** [comment]
I think you can pet them even if they're hostile
It's just not a good idea

**Comment by clusterfack** [comment]
My CARP ARMADA WILL DESTROY THE VOX

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -12,7 +12,9 @@
 	var/projectilesound
 	var/casingtype
 	var/move_to_delay = 2 //delay for the automated movement.
-	var/list/friends = list()
+
+	var/list/friends = list() //List of friends. Accepts both paths (/mob/living/simple_animal/corgi) and mob references.
```
Comment (clusterfack) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: Oh man this holds actual mob references? That ain't good it'll break garbage collection for them. Using text refs would be preferrable if possible.

Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:16]: Hold the DNA string instead

Comment (gbasood) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:16]: Do simple animals have DNA?

Comment (unid15) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:16]: Nope

Comment (PJB3005) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm:16]: Nope.

All carbons have a var for it though (simple animals aren't carbon).

**Comment by gbasood** [comment]
Did you know? THERE WAS NEVER A MAX LIMIT ON CARPS

**Comment by Exxion** [comment]
CARP THE HUMANS RACE WAR NOW

**Comment by Exxion** [comment]
when

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 2]
```diff
@@ -146,7 +146,16 @@
...
+
+					//The following three lines allow baby carps to start seeing ALL humans / vox / monkeys / etc as friends.
+					//If only one human feeds a carp, the carp will only see that human as a friend.
+					//If, however, two or more humans feed a carp, the carps will see the entire human species as friends. They'll still be agressive towards vox (unless two or more vox feed them) and other mobs.
+					for(var/mob/M in friends)
+						if(M.type == user.type)
+							friends[user.type] += 2
+							break
+
 					friends.Add(user)
```
Comment (gbasood) [inline code/modules/mob/living/simple_animal/hostile/carp.dm:159]: Uh
Carp still storing references to player mobs who will need to be qdel'd when they explode horribly = bad

**Comment by Probe1** [comment]
RIP in progress

**Comment by ghost** [comment]
All dreams must die.

**Comment by ComicIronic** [comment]
RIP this.

**Pull request closed without merging**